### DISCUSSION TITLE
Why are the Weather intervals (week, month, year) all off by one day?

### DISCUSSION THREAD

**Discussion by mzurhorst** [Daveiano/weewx-wdc#115] [General]
Hi.

I am wondering why the website tells me that the intervall for the current months spans "01.02.2023 - 01.03.2023"? -- Same for year, which goes to Jan 1, 2024 actually.

Whilst being a minor glitch, I do not understand this design decision at all since it is uncommon.
Can somebody explain which this is done this way?  Plus, is this even theme-related, or would that be a discussion for the weewx project itself?

Thanks a lot.

Best regards, 
  Marcus

**Comment by Daveiano** [comment]
Hi and thank you for your input.

You are right, the time spans are displayed as one day off. That's a display issue of the skin when showing the date spans. To be honest, I noticed this bug while developing but was ignoring it, until I forgot about it. You are the first one to notice/report this. 

I will put this on the plan for the next release!

**Reply by mzurhorst** [reply]
Apologies for reminding you on this one.  And thanks for your work!

**Comment by Daveiano** [comment]
I actually got time to look into this one and I found the following in the weewx-docs, from https://weewx.com/docs/customizing.htm#Start,_end,_and_dateTime:

> For aggregation periods, such as $month, you can request the start, end, or length of the period, by using suffixes .start, .end, or .length, respectively. For example,
> `The current month runs from $month.start to $month.end and has $month.length.format("%(day)d %(day_label)s").`
> results in
> The current month runs from 01/01/2010 12:00:00 AM to 02/01/2010 12:00:00 AM and has 31 days.
> In addition to the suffixes .start and .end, the suffix .dateTime is provided for backwards compatibility. Like .start, it refers to the start of the interval.

The skin already uses `$period.start` and `$period.end` as suggested by the weewx docs, so this seems to be the default weewx behaviour. I tend to not change anything about that because it's just weewx default templating tags. Someone could post a question to the weewx user group about this, I guess.

Sorry about that, I thought I was using something custom to show the periods, but it's just the default way (in my understanding).

**Reply by mzurhorst** [reply]
Thanks a lot for explanations, I have now asked in the user group:   https://groups.google.com/g/weewx-user/c/9dcSu4MQ7a0

**Comment by Daveiano** [comment]
Fixed in https://github.com/Daveiano/weewx-wdc/commit/20b6f62dbfd83f7aa9be14394d9caee22cf5615c

Will be available in v3.0.0 of the skin.

**Reply by mzurhorst** [reply]
Wow, thanks a lot.
I did not expect this at all when I saw the discussion progressing into the direction that my expectation would be wrong.
